azext_iot/iothub/providers/message_endpoint.py

```diff
@@ -0,0 +1,351 @@
+                    namespace_name=endpoint_account_name,
+                    queue_name=entity_path,
+                    policy_name=endpoint_policy_name,
+                    rg=endpoint_resource_group,
+                    sub=endpoint_subscription_id
+                )
+            new_endpoint["entityPath"] = entity_path
+            endpoints.service_bus_queues.append(new_endpoint)
+        elif EndpointType.ServiceBusTopic.value == endpoint_type.lower():
+            if fetch_connection_string:
+                new_endpoint["connectionString"] = get_servicebus_topic_cstring(
+                    cmd=self.cli,
+                    namespace_name=endpoint_account_name,
+                    topic_name=entity_path,
+                    policy_name=endpoint_policy_name,
+                    rg=endpoint_resource_group,
+                    sub=endpoint_subscription_id
+                )
+            new_endpoint["entityPath"] = entity_path
+            endpoints.service_bus_topics.append(new_endpoint)
+        elif EndpointType.CosmosDBContainer.value == endpoint_type.lower():
+            if fetch_connection_string:
+                # try to get connection string
+                new_endpoint["connectionString"] = get_cosmos_db_cstring(
+                    cmd=self.cli,
+                    account_name=endpoint_account_name,
+                    rg=endpoint_resource_group,
+                    sub=endpoint_subscription_id
+                )
+            if connection_string:
+                # parse out key from connection string
+                if not primary_key and not secondary_key:
+                    parsed_cs = parse_cosmos_db_connection_string(connection_string)
+                    primary_key = parsed_cs["AccountKey"]
+                    secondary_key = parsed_cs["AccountKey"]
+                # parse out endpoint uri from connection string
+                if not endpoint_uri:
+                    new_endpoint["endpointUri"] = parsed_cs["AccountEndpoint"]
+            if authentication_type != AuthenticationType.IdentityBased.value and not any([primary_key, secondary_key]):
+                raise RequiredArgumentMissingError(
+                    "Primary key via --primary-key, secondary key via --secondary-key, or connection string via "
+                    "--connection-string is required."
+                )
+            if primary_key and not secondary_key:
+                secondary_key = primary_key
+            if secondary_key and not primary_key:
+                primary_key = secondary_key
+            if not new_endpoint["endpointUri"]:
+                raise RequiredArgumentMissingError(
+                    "Endpoint uri via --endpoint-uri or connection string via --connection-string is required."
+                )
+            if partition_key_name and not partition_key_template:
+                partition_key_template = '{deviceid}-{YYYY}-{MM}'
+            del new_endpoint["connectionString"]
```

Comment:
This value is deleted before it's used.

Reply:
yes cause only cosmos db doesn't want a connection string. I use the connection string to get the primary key/symmetric key and uri props instead